src/dev/impl/DevToys/Helpers/XmlHelper.cs

```diff
@@ -0,0 +1,103 @@
+﻿#nullable enable
+
+using System;
+using System.IO;
+using System.Text;
+using System.Xml;
+using DevToys.Core;
+using DevToys.Models;
+
+namespace DevToys.Helpers
+{
+    internal static class XmlHelper
+    {
+        /// <summary>
+        /// Detects whether the given string is a valid Xml or not.
+        /// </summary>
+        internal static bool IsValid(string? input)
+        {
+            if (string.IsNullOrWhiteSpace(input))
+            {
+                return false;
+            }
+
+            input = input!.Trim();
+
+            if (input.Length < 4)
+            {
+                return false;
+            }
+
+            if (input.StartsWith("<") && input.EndsWith(">"))
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Format a string to the specified Xml format.
+        /// </summary>
+        internal static string Format(string? input, Indentation indentationMode, bool newLineOnAttributes)
+        {
+            if (!IsValid(input))
+            {
+                return string.Empty;
+            }
+
+            try
+            {
+                var xmlDocument = new XmlDocument();
+                xmlDocument.Load(new StringReader(input));
+
+                var xmlWriterSettings = new XmlWriterSettings()
+                {
+                    Async = true,
+                    OmitXmlDeclaration = true,
```

Comment:
Why omitting the XML declaration? Would it be possible to conserve the XML declaration only if it exists in the input too?

Reply:
Somehow the encoding is always UTF-16, no matter the input. Omit the XML declaration if it did not exist on the input string.

Reply:
That's probably fine for now. Let's ship it this way and see if we get user feedback 🤔